src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -7059,8 +7018,9 @@ void func_808417FC(Player* this, GlobalContext* globalCtx) {
 
 void func_80841860(GlobalContext* globalCtx, Player* this) {
     f32 frame;
-    LinkAnimationHeader* sp38 = D_80853914[this->modelAnimType + 144];
-    LinkAnimationHeader* sp34 = D_80853914[this->modelAnimType + 150];
+    // fake match? see func_80833664
+    LinkAnimationHeader* sp38 = D_80853914[0][this->modelAnimType + PLAYER_ANIMGROUP_24 * ARRAY_COUNT(D_80853914[0])];
+    LinkAnimationHeader* sp34 = D_80853914[0][this->modelAnimType + PLAYER_ANIMGROUP_25 * ARRAY_COUNT(D_80853914[0])];
```

Comment:
Isn't that also UB?
I think we should probably make it a 1D array if we don't have a choice, and perhaps with a macro to simplify the code when accessing it (to avoid the * PLAYER_ANIMTYPE_MAX everywhere)

Reply:
Yeah I agree. I guess I'll change it to a 1D array. I wasn't sure if this code is used at all or of any importance but still it should probably be a 1D array

---------

Unrelated to this PR but same issue, should the sSetupDL being 2D with this 1D cast also be reverted to 1D

https://github.com/zeldaret/oot/blob/054bac758306026b99358f97ac22062fbedff16b/src/code/z_rcp.c#L901

Reply:
Yeah you're right about z_rcp.c having the same issue. It was changed a while ago, but tbh I'm still not sure whether it's an improvement. I guess we should discuss it separately